src/app/clusters/temperature-control-server/temperature-control-server.cpp

```diff
@@ -0,0 +1,186 @@
+        if (temperatureLevelOptionsProvider.begin() == nullptr)
+        {
+            aEncoder.EncodeEmptyList();
+            return CHIP_NO_ERROR;
+        }
+        CHIP_ERROR err;
+        err = aEncoder.EncodeList([temperatureLevelOptionsProvider](const auto & encoder) -> CHIP_ERROR {
+            const auto * end = temperatureLevelOptionsProvider.end();
+            for (auto * it = temperatureLevelOptionsProvider.begin(); it != end; ++it)
+            {
+                auto & temperatureLevelOption = *it;
+                ReturnErrorOnFailure(encoder.Encode(temperatureLevelOption));
+            }
+            return CHIP_NO_ERROR;
+        });
+        ReturnErrorOnFailure(err);
+    }
+    return CHIP_NO_ERROR;
+}
+
+bool TemperatureControlHasFeature(EndpointId endpoint, TemperatureControl::Feature feature)
+{
+    bool success;
+    uint32_t featureMap;
+    success = (Attributes::FeatureMap::Get(endpoint, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
+
+    return success ? ((featureMap & to_underlying(feature)) != 0) : false;
+}
+
+/**********************************************************
+ * Callbacks Implementation
+ *********************************************************/
+
+bool emberAfTemperatureControlClusterSetTemperatureCallback(app::CommandHandler * commandObj,
+                                                            const app::ConcreteCommandPath & commandPath,
+                                                            const Commands::SetTemperature::DecodableType & commandData)
+{
+    auto & targetTemperature      = commandData.targetTemperature;
+    auto & targetTemperatureLevel = commandData.targetTemperatureLevel;
+    EndpointId endpoint           = commandPath.mEndpointId;
+    Status status                 = Status::Success;
+    EmberAfStatus emberAfStatus   = EMBER_ZCL_STATUS_SUCCESS;
+
+    if (TemperatureControlHasFeature(endpoint, Feature::kTemperatureNumber) &&
+        TemperatureControlHasFeature(endpoint, Feature::kTemperatureLevel))
+    {
+        commandObj->AddStatus(commandPath, Status::Failure);
+        return false;
+    }
+
+    if (TemperatureControlHasFeature(endpoint, TemperatureControl::Feature::kTemperatureNumber))
+    {
+        if (targetTemperature.HasValue())
+        {
+            int16_t minTemperature = 0;
+            int16_t maxTemperature = 0;
+            emberAfStatus          = MinTemperature::Get(endpoint, &minTemperature);
+            if (emberAfStatus != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                status = app::ToInteractionModelStatus(emberAfStatus);
+            }
+
+            emberAfStatus = MaxTemperature::Get(endpoint, &maxTemperature);
+            if (emberAfStatus != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                status = app::ToInteractionModelStatus(emberAfStatus);
+            }
+
+            if (targetTemperature.Value() < minTemperature || targetTemperature.Value() > maxTemperature)
```

Comment:
We are reaching this code when minTemperature might be garbage.

Please work through the logic here and further on carefully.

I did not read past this point.

Reply:
Handeled the case